daemons/execd/pacemaker-execd.h

```diff
@@ -41,7 +41,14 @@ typedef struct lrmd_rsc_s {
      * that have been handed off from the pending ops list. */
     GList *recurring_ops;
 
-    int st_probe_rc; // What value should be returned for a probe if stonith
+    /* If this resource is a fence device, probes are handled internally by the
+     * executor, and this value indicates the result that should currently be
+     * returned for probes. It should be one of:
+     * PCMK_EXEC_DONE (to indicate "running"),
+     * PCMK_EXEC_NO_FENCE_DEVICE ("not running"), or
+     * PCMK_EXEC_NOT_CONNECTED ("unknown because fencer connection was lost").
+     */
+    enum pcmk_exec_status fence_probe_result;
```

Comment:
name is maybe a bit misleading as it is just the exec_status part of the result.
not sure though if it would make sense to go for a full result here.
we wouldn't need the logic in finalize_fence_device_probe anymore and as
a consequence could toss finalize_fence_device_probe entirely.

Reply:
I like that idea, but I'll save it for a follow-up